libs/dh/shared/feature-authorization/src/lib/actor-token.service.spec.ts

```diff
@@ -0,0 +1,217 @@
+/**
+ * @license
+ * Copyright 2020 Energinet DataHub A/S
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License2");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ActorTokenService } from './actor-token.service';
+import {
+  MarketParticipantUserHttp,
+  TokenHttp,
+} from '@energinet-datahub/dh/shared/domain';
+import {
+  HttpHandler,
+  HttpRequest,
+  HttpHeaders,
+  HttpResponse,
+} from '@angular/common/http';
+import { firstValueFrom, of } from 'rxjs';
+
+describe(ActorTokenService.name, () => {
+  const createActorsRequest = () =>
+    new HttpRequest<string>(
+      'GET',
+      'https://localhost:5000/v1/MarketParticipantUser/Actors',
+      {
+        headers: new HttpHeaders({ Authorization: 'Bearer xyz' }),
+      }
+    );
+
+  test('should return true if request is for getting actors', async () => {
+    // arrange
+    const request = createActorsRequest();
+
+    const target = new ActorTokenService(
+      {} as MarketParticipantUserHttp,
+      {} as TokenHttp
+    );
+
+    // act
+    const actual = target.isPartOfAuthFlow(request);
+
+    // assert
+    expect(actual).toEqual(true);
+  });
+
+  test('should return true if request is for getting token', async () => {
+    // arrange
+    const request = new HttpRequest<string>(
+      'GET',
+      'https://localhost:5000/v1/Token'
+    );
+
+    const target = new ActorTokenService(
+      {} as MarketParticipantUserHttp,
+      {} as TokenHttp
+    );
+
+    // act
+    const actual = target.isPartOfAuthFlow(request);
+
+    // assert
+    expect(actual).toEqual(true);
+  });
+
+  test('should return false if request is not for authentication flow', async () => {
+    // arrange
+    const request = new HttpRequest<string>(
+      'GET',
+      'https://localhost:5000/v1/Not/Relevant'
+    );
+
+    const target = new ActorTokenService(
+      {} as MarketParticipantUserHttp,
+      {} as TokenHttp
+    );
+
+    // act
+    const actual = target.isPartOfAuthFlow(request);
+
+    // assert
+    expect(actual).toEqual(false);
+  });
+
+  test('should return false if request is not for internal authentication flow', async () => {
+    // arrange
+    const request = new HttpRequest<string>(
+      'GET',
+      'https://dev002datahubb2c.b2clogin.com/dev002datahubb2c.onmicrosoft.com/b2c_some_policy/oauth2/v2.0/token'
```

Comment:
nitpick: the `isPartOfAuthFlow` checks whether the request url ends with a specific value. The origin is not checked. So in theory if `dev002datahubb2c.b2clogin.com` had an endpoint that ends with `/v1/Token` (or the other endpoint), the `isPartOfAuthFlow` would return `true`. Though veeery unlikely to happen. :)

Reply:
We have a test (should return false if request is not for internal authentication flow) checking that `b2clogin.com`-url is rejected. But the code is the second line of defense, as the endpoint is sent without Bearer header. So it is currently impossible to reach `isPartOfAuthFlow`.